shared/constants/chat2/message.tsx

```diff
@@ -19,6 +19,14 @@ import {assertNever} from '../../util/container'
 import invert from 'lodash/invert'
 import shallowEqual from 'shallowequal'
 
+export const getMessageStateExtras = (state: TypedState, conversationIDKey: Types.ConversationIDKey) => {
+  const getLastOrdinal = () =>
```

Comment:
this is a callback so we don't actually have to run it all the time. its only used for the pending messages